doc/manual/src/command-ref/files/profiles.md

```diff
@@ -1,17 +1,42 @@
-## Profiles
+## Profile Directory
 
-A directory that contains links to profiles managed by [`nix-env`] and [`nix profile`]:
+A directory that contains profiles.
+Profiles are typically managed by [`nix-env`] and [`nix profile`].
+These are the default profile directories:
 
 - `$XDG_STATE_HOME/nix/profiles` for regular users
 - `$NIX_STATE_DIR/profiles/per-user/root` if the user is `root`
 
-A profile is a directory of symlinks to files in the Nix store.
+## Profile
+
+Profiles are a basic version control mechanism built atop more core Nix constructs and symlinks.
+Profiles consiste of
+
+- An map of zero or more generations, each mapping a *generation number* to a [store object](@docroot@/glossary.md#gloss-store-object).
+
+- An optional *current* or *active* generation, which is one of those in the map.
+
+Profiles are represented on disk by symlinks within a directory that obey a certain structure.
+Profiles have names, which allows storing more than one profile in the same directory.
+The symlinks of a single profile all begin with the name of profile, which identifies which symlinks belong to which profile within a directory.
+A combined `<profile-directory>/<profile-name>` path uniquely identifies a profile on the local file system, by specifying both its location and name.
+
+Each generation is represented by a symlink with a name with 3 dash-separated components: `<profile-name>-<generation-number>-link`.
```

Comment:
Shouldn't this be in the "filesystem layout" subsection?

Reply:
I think I mean that to just be "example" really, as per the above I think this whole chapter is "filesystem layout".